Replicator/tests/ReplParams.cc

```diff
@@ -20,6 +20,24 @@ ReplParams::ReplParams(const std::vector<C4ReplicationCollection>& collections_)
     socketFactory = nullptr;
 }
 
+ReplParams::ReplParams(const std::vector<C4CollectionSpec> &collSpecs, C4ReplicatorMode push, C4ReplicatorMode pull) {
+    _collectionVector = std::vector<C4ReplicationCollection>{ collSpecs.size() };
+    for(int i = 0; i < collSpecs.size(); ++i) {
+        _collectionVector[i] = { collSpecs[i], push, pull };
+    }
+    collections = _collectionVector.data();
+    collectionCount = _collectionVector.size();
+    _optionsDict = {};
+    optionsDictFleece = _optionsDict.data();
+    onStatusChanged = nullptr;
+    onDocumentsEnded = nullptr;
+    onBlobProgress = nullptr;
+    propertyDecryptor = nullptr;
+    propertyEncryptor = nullptr;
+    callbackContext = nullptr;
+    socketFactory = nullptr;
```

Comment:
This is too noisy.  You can define default values for these null items in the header and there is no need to explicitly set them all to null here.

Reply:
It extends from a C-style struct, which is where the values come from. Is there a way to still have default values?

Reply:
Hmmm, well if the header is also used form C then it will be more trouble than it is worth.  If the header is C++ only then it doesn't matter and you can do it.  Or you can use the modern C++ initialization

```
Constructor()
    :field1(value1)
    ,field2(value2)
{ }
```

Reply:
Seems I can't use that type of initialisation for the C-struct values, but now I've fixed my IDE being slow, I found out I can just do `: C4ReplicatorParameters()` and all the struct values get initialised.